predicate.go

```diff
@@ -0,0 +1,219 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"io"
+	"net/http"
+
+	"github.com/golang/glog"
+	"k8s.io/api/core/v1"
+	"k8s.io/api/extensions/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	schedulerapi "k8s.io/kubernetes/pkg/scheduler/api/v1"
+)
+
+// Predicates implemented filter functions.
+// The filter list is expected to be a subset of the supplied list.
+func Predicates(clientset *kubernetes.Clientset) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		var buf bytes.Buffer
+		body := io.TeeReader(r.Body, &buf)
+		glog.V(2).Infof("body: %s", buf.String())
+
+		var extenderArgs schedulerapi.ExtenderArgs
+		var extenderFilterResult *schedulerapi.ExtenderFilterResult
+
+		if err := json.NewDecoder(body).Decode(&extenderArgs); err != nil {
+			extenderFilterResult = &schedulerapi.ExtenderFilterResult{
+				Nodes:       nil,
+				FailedNodes: nil,
+				Error:       err.Error(),
+			}
+		} else {
+			extenedResourceScheduler := &ExtendedResourceScheduler{
+				Clientset: clientset,
+			}
+			extenderFilterResult = filter(extenderArgs, extenedResourceScheduler)
+		}
+
+		w.Header().Set("Content-Type", "application/json")
+		if resultBody, err := json.Marshal(extenderFilterResult); err != nil {
+			w.WriteHeader(http.StatusInternalServerError)
+			w.Write([]byte(err.Error()))
+		} else {
+			w.WriteHeader(http.StatusOK)
+			w.Write(resultBody)
+		}
+	}
+}
+
+// Bind delegates the action of binding a pod to a node to the extender.
+func Bind(clientset *kubernetes.Clientset) http.HandlerFunc {
```

Comment:
do we really need this ?

Reply:
I have removed bind and tested it, and then scheduler looks work fine. I will read some code about bind to know how does it work later.